heron/instance/src/java/com/twitter/heron/instance/Slave.java

```diff
@@ -62,6 +74,13 @@ public Slave(SlaveLooper slaveLooper,
     this.streamInCommunicator = streamInCommunicator;
     this.streamOutCommunicator = streamOutCommunicator;
     this.inControlQueue = inControlQueue;
+    this.metricsOutCommunicator = metricsOutCommunicator;
+    this.serializer =
+        SerializeDeSerializeHelper.getSerializer(helper.getTopologyContext().getTopologyConfig());
+
+    // TODO (nlu): Create the state with corresponding type according to config
+    instanceState = new HashMapState<>();
```

Comment:
But the state should always be injected from the state server right? The slave should not assume that the state is empty state until it hears from the state server. It should be told this state by someone else, even if it's empty.